orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -509,6 +520,25 @@ class JedisExecutionRepository implements ExecutionRepository {
     return currentObservable
   }
 
+  @Override
+  Orchestration retrieveOrchestrationForCorrelationId(@Nonnull String correlationId) throws ExecutionNotFoundException {
+    String key = "correlation:$correlationId"
+    withJedis(getJedisPoolForId(key)) { Jedis correlationJedis ->
+      def orchestrationId = correlationJedis.get(key)
+
+      if (orchestrationId != null) {
+        def orchestration = withJedis(getJedisPoolForId(orchestrationId)) { Jedis jedis ->
+          retrieveInternal(jedis, Orchestration, orchestrationId)
+        }
+        if (!orchestration.status.isComplete()) {
+          return orchestration
+        }
+        correlationJedis.del(key)
+      }
+      throw new ExecutionNotFoundException("No Orchestration found for correlation ID $correlationId")
```

Comment:
Is this really what we want to do rather than returning `null`?

Reply:
I was keeping pattern with the other find methods. I'm totally down with (and would prefer) using null, but throwing an exception seemed consistent with the rest of the repository.

Reply:
Fair enough